src/Nav.js

```diff
@@ -1,359 +1,233 @@
+    /**
+     * @default 'nav'
+     */
+    bsPrefix: PropTypes.string,
+
+    /**
+     * The visual variant of the nav items.
+     *
+     * @type {('tabs'|'pills')}
+     */
+    variant: PropTypes.string,
+
+    /**
+     * Marks the NavItem with a matching `eventKey` as active. Has a
+     * higher precedence over `activeHref`.
+     */
+    activeKey: PropTypes.any,
 
-    activeNode.firstChild.focus();
+    /**
+     * Have all `NavItem`s to proportionatly fill all available width.
+     */
+    fill: PropTypes.bool,
+
+    /**
+     * Have all `NavItem`s to evenly fill all available width.
+     *
+     * @type bool
+     */
+    justify: all(
+      PropTypes.bool,
+      ({ justify, navbar }) =>
+        justify && navbar
+          ? Error('justify navbar `Nav`s are not supported')
+          : null
+    ),
+
+    /**
+     * A callback fired when a NavItem is selected.
+     *
+     * ```js
+     * function (
+     *  Any eventKey,
+     *  SyntheticEvent event?
+     * )
+     * ```
+     */
+    onSelect: PropTypes.func,
+
+    /**
+     * ARIA role for the Nav, in the context of a TabContainer, the default will
+     * be set to "tablist", but can be overridden by the Nav when set explicitly.
+     *
+     * When the role is set to "tablist" NavItem focus is managed according to
+     * the ARIA authoring practices for tabs:
+     * https://www.w3.org/TR/2013/WD-wai-aria-practices-20130307/#tabpanel
+     */
+    role: PropTypes.string,
+
+    /**
+     * Apply styling an alignment for use in a Navbar. This prop will be set
+     * automatically when the Nav is used inside a Navbar.
+     */
+    navbar: PropTypes.bool,
+
+    componentClass: elementType,
+
+    /** @private */
+    onKeyDown: PropTypes.func
+  };
+
+  static defaultProps = {
+    justify: false,
+    fill: false,
+    componentClass: 'ul'
+  };
+
+  static getDerivedStateFromProps(
+    { activeKey, getControlledId, getControllerId, role, onSelect },
+    prevState
+  ) {
+    return {
+      ...prevState,
+      navContext: {
+        role, // used by NavLink to determine it's role
+        onSelect,
+        activeKey,
+        getControlledId: getControlledId || noop,
+        getControllerId: getControllerId || noop
+      }
+    };
   }
 
-  getActiveProps() {
-    const tabContainer = this.context.$bs_tabContainer;
+  constructor(...args) {
+    super(...args);
 
-    if (tabContainer) {
-      warning(
-        this.props.activeKey == null && !this.props.activeHref,
-        'Specifying a `<Nav>` `activeKey` or `activeHref` in the context of ' +
-          'a `<TabContainer>` is not supported. Instead use `<TabContainer ' +
-          `activeKey={${this.props.activeKey}} />\`.`
-      );
+    this.state = { navContext: null };
+  }
 
-      return tabContainer;
-    }
+  componentDidUpdate() {
+    if (!this._needsRefocus || !this.listNode) return;
 
-    return this.props;
+    let activeChild = qsa(this.listNode, `.nav-link.active`).pop();
```

Comment:
this seems like an awkward way to just do `querySelector` :p

Reply:
maybe just `.active`? hard-coding `.nav-link` seems wrong if `bsPrefix` is customizable

Reply:
ah true, 👍